### PR TITLE
[fider/581] Invalid posts should return a 404 and not a 500

### PR DESCRIPTION
**Pull request by peterver**
**Issue:**  #581 

- Adjust the handlers to send back a 404 if the `ParamAsInt("number")` validation fails.
- Add an endpoint to retrieve a post (Rest)

### CODE REVIEW

File: app/cmd/routes.go [hunk 1]
```diff
@@ -159,6 +159,7 @@ func routes(r *web.Engine) *web.Engine {
 		api.Use(middlewares.IsAuthenticated())
 
 		api.Post("/api/v1/posts", apiv1.CreatePost())
+                api.Get("/api/v1/posts/:number", apiv1.GetPost())
```
Comment (goenning) [inline app/cmd/routes.go]: Wow, how did I miss this API? Thank you!

Few comments:

1. We can move it before the IsAuthenticated middleware, right next to  `/api/v1/posts`
2. Align it with the rest of the code.

File: app/handlers/apiv1/post.go [hunk 2]
```diff
@@ -52,6 +52,23 @@ func CreatePost() web.HandlerFunc {
 	}
 }
 
+// GetPost retrieves the existing post by id
```
Comment (goenning) [inline app/handlers/apiv1/post.go]: id -> number

// GetPost retrieves the existing post by number

**Comment by peterver** [comment]
@goenning will take a look at the roc tomorrow :) !

**Comment by peterver** [comment]
@goenning changes are done :)

**Comment by peterver** [comment]
@goenning I also added a test for the new GetPost route

**Comment by goenning** [comment]
@peterver  Add a `Expect(query.String("title")).Equals("My First Post")` assertion to that query :)

**Comment by goenning** [comment]
@peterver one more thing, you need to move the `api.Get("/api/v1/posts/:number", apiv1.GetPost())` before the `api.Use(middlewares.IsAuthenticated())`. Posts are public and without that only authenticated users would be able to use that API.

**Comment by peterver** [comment]
there ya go @goenning :)

**Comment by goenning** [comment]
Approved! But looks like CircleCI didn't run the build :(

**Comment by peterver** [comment]
@goenning probably stuck in a queue somewhere :/. Anyway to retrigger a circleci build or let's just way for the queue to do its thing ?

**Comment by goenning** [comment]
I don't see any queue on CircleCI. Can you change the title or description on that test and push another commit? That should force a new build.

**Comment by peterver** [comment]
@goenning There we go, now circle ci is rollin :)

**Comment by goenning** [comment]
That's merged, thank you!

I'll prepare and cut the 0.16.0 release now! 🎆

**Pull request merged**